CMakeLists.txt

```diff
@@ -158,7 +158,26 @@ set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 
 
 
-find_package(ZeroMQ 4 REQUIRED)
+#From: https://github.com/zeromq/cppzmq/
+if (NOT TARGET libzmq)
+  find_package(ZeroMQ 4 QUIET)
+
+  # libzmq autotools install: fallback to pkg-config
+  if(NOT ZeroMQ_FOUND)
```

Comment:
I have never seen this pattern before so I am not sure. My understanding is that the default strategy of CMake is to  look for `FindXxx.cmake` (module mode),  then look for CMake package first (config mode). Then use `pkg-config` at last resort.

The priority can be modified with [`CMAKE_FIND_PACKAGE_PREFER_CONFIG`](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html) from CMake v3.15.

Reply:
I hadn't seen this strategy as well but this is what cppzmq uses which seems to a fairly popular project. 
https://github.com/zeromq/cppzmq/blob/master/cppzmqConfig.cmake.in

Not saying that being popular automatically means good quality code =) 

Playing around, what I appreciate is that if the ZeroMQ installation provides a ZeroMQConfig.cmake the first find_package call works (which is the case for the conda zmq but not my system lib) and otherwise it uses the fallback strategy. 

But if you have nice FindZeroMQ.cmake we can try using that instead...

Reply:
> Not saying that being popular automatically means good quality code =)

But being consistant with a popular project is a good idea!

I guess `CMAKE_FIND_PACKAGE_PREFER_CONFIG` is the new way to do this (reverse the  priority of the `find_package` strategy), but requiring CMake v3.15 can be an issue for a popular project.